src/components/BugsTable/index.jsx

```diff
@@ -0,0 +1,292 @@
+        ? this.state.data.sort((a, b) => (b[orderBy] < a[orderBy] ? -1 : 1))
+        : this.state.data.sort((a, b) => (a[orderBy] < b[orderBy] ? -1 : 1));
+
+    this.setState({ data, order, orderBy });
+  };
+
+  handleChangePage = (event, page) => {
+    const threshold = 40;
+    const { rowsPerPage, data, hasNextPageList } = this.state;
+
+    this.setState({ page });
+
+    if (
+      data.length - rowsPerPage * page < threshold &&
+      hasNextPageList.length > 0
+    ) {
+      this.fetchDataNext(hasNextPageList);
+    }
+  };
+
+  handleChangeRowsPerPage = event => {
+    this.setState({ rowsPerPage: event.target.value });
+  };
+
+  isSelected = id => this.state.selected.indexOf(id) !== -1;
+
+  render() {
+    const { classes } = this.props;
+    const {
+      data,
+      order,
+      orderBy,
+      rowsPerPage,
+      page,
+      loading,
+      error,
+    } = this.state;
+    const emptyRows =
+      rowsPerPage - Math.min(rowsPerPage, data.length - page * rowsPerPage);
+
+    return (
+      <Fragment>
+        {loading && <div>loading...</div>}
+        {error ? (
+          <div>error...</div>
+        ) : (
+          <Paper className={classes.root}>
+            <BugsTableToolbar />
+            <div className={classes.tableWrapper}>
+              <Table className={classes.table} aria-labelledby="tableTitle">
+                <BugsTableHead
+                  order={order}
+                  orderBy={orderBy}
+                  onRequestSort={this.handleRequestSort}
+                  rowCount={data.length}
+                />
+                <TableBody>
+                  {data
+                    .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
+                    .map(n => (
+                      <BugsTableEntry
+                        key={n.id}
+                        project={n.project}
+                        description={n.description}
+                        tag={n.tag}
+                        assignedto={n.assignedto}
+                        lastupdate={n.lastupdate}
+                      />
+                    ))}
+                  {emptyRows > 0 && (
+                    <TableRow style={{ height: 49 * emptyRows }}>
+                      <TableCell colSpan={6} />
+                    </TableRow>
+                  )}
+                </TableBody>
+              </Table>
+            </div>
+            <TablePagination
+              component="div"
+              labelDisplayedRows={({ from, to }) => `${from}-${to}`}
+              count={data.length}
+              rowsPerPage={rowsPerPage}
+              page={page}
+              backIconButtonProps={{
+                'aria-label': 'Previous Page',
+              }}
+              nextIconButtonProps={{
+                'aria-label': 'Next Page',
+              }}
+              onChangePage={this.handleChangePage}
+              onChangeRowsPerPage={this.handleChangeRowsPerPage}
+            />
+          </Paper>
```

Comment:
You can create a component `ConnectionDataTable` that uses a connection to render a table. I think you can use most of https://github.com/taskcluster/taskcluster-web/blob/master/src/components/ConnectionDataTable/index.jsx.

Reply:
No, you can keep it in this pull-request because this component will only work with a graphql connection as input.